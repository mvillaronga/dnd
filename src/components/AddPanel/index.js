import React, { useRef } from 'react'
import { useDispatch } from 'react-redux'

import { addNpc } from '../../redux/npcs'

const AddPanel = () => {
    const inputName = useRef()
    const dispatch = useDispatch()
    
    const onButtonClick = () => {
         dispatch(addNpc(inputName.current.value))
         inputName.current.value = ''
         inputName.current.focus()
     }

    return <div>
        Npc &nbsp;
        <input type="text" ref={inputName} /> &nbsp;
        <button onClick={onButtonClick}>Add</button>
    </div>
}

export default AddPanel;