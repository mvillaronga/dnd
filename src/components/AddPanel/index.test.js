import React from 'react'
import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure, mount } from 'enzyme';

import AddPanel from '../AddPanel'

import npcSlice from '../../redux/npcs'

configure({ adapter: new Adapter() });

const entered = 'Gandalf'

describe('AddPanel tests', () => {
  const mockStore = configureMockStore([thunk])
  const store = mockStore({})

  beforeEach(() => {
    store.clearActions();
  })

  it('should render the panel', () => {
    const comp = shallow(
      <Provider store={store}>
        <AddPanel />
      </Provider>
    )

    expect(comp.html()).toMatchSnapshot()
  })

  it('should fire the addNPC event with the entered value on click', () => {
    const comp = mount(
      <Provider store={store}>
        <AddPanel />
      </Provider>
    )

    comp.find("input").getDOMNode().value = entered
    comp.find("button").first().simulate('click')
    expect(store.getActions()).toEqual([{ 
        type: npcSlice.actions.addNpc.type,
        payload: { id: 0, name: 'Gandalf', selected: false }
      }]
    )
    expect(comp.find("input").getDOMNode().value).toEqual('')
  })
})