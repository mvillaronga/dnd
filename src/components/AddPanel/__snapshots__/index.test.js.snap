// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPanel tests should render the panel 1`] = `"<div>Npc  <input type=\\"text\\"/>  <button>Add</button></div>"`;
